packages/hint-manifest-icons/src/hint.ts

```diff
@@ -267,6 +270,73 @@ export default class ManifestIconHint implements IHint {
             return validSizes;
         };
 
+        const validatePurposesValues = (purposes: string[], resource: string, iconIndex: number, getLocation: JSONLocationFunction) => {
+            for (const purpose of purposes) {
+                // Check if the purpose is valid.
+                if (!validPurposes.includes(purpose)) {
+                    context.report(resource, getMessage('iconPurposeInvalid', context.language, purpose), {
+                        location: getLocation(`icons[${iconIndex}].purpose`),
+                        severity: Severity.warning
+                    });
+                }
+            }
+        };
+
+        const validateDuplicatedValues = (purposes: string[], resource: string, iconIndex: number, getLocation: JSONLocationFunction) => {
+            /*
+             * Creating a set will remove automatically the duplicated items.
+             */
+            const purposesSet = new Set(purposes);
+
+            if (purposes.length === purposesSet.size) {
+                // No duplicated found.
+                return;
+            }
+
+            const purposesCopy = [...purposes];
+
+            for (const purpose of purposesSet) {
+                const purposeIndex = purposesCopy.indexOf(purpose);
+
+                if (purposeIndex >= 0) {
+                    purposesCopy.splice(purposeIndex, 1);
+                }
+            }
+
+            /*
+             * Check if purposesCopy contains any purpose, if so, those
+             * items are duplicated.
+             */
+            if (purposesCopy.length > 0) {
+                context.report(resource, getMessage('iconPurposeDuplicate', context.language, [...(new Set(purposesCopy))].join(' ')), {
```

Comment:
Probably should comma-separate these:
```suggestion
                context.report(resource, getMessage('iconPurposeDuplicate', context.language, [...(new Set(purposesCopy))].join(', ')), {
```